feat: enable custom configurations for LimeSurvey service

mariajgrimaldi opened the pull request:
### Description
This PR adds a configuration file during the image-building process so it can be used by the [LimeSurvey docker entrypoint](https://github.com/adamzammit/limesurvey-docker/blob/master/docker-entrypoint.sh#L58) when starting the service to populate its configurations. The base file config.php was copied from a remote installation and then modified by appending:

```
         'request' => array(
			'enableCsrfValidation'=>true,
			'csrfCookie' => array(
				'sameSite' => 'None',
				'secure' => true,
			),
		),
```
and
```
 'x_frame_options' => 'allow',
```
So the LimeSurvey installations allow being embedded in domains different than theirs.

This PR solves this customization issue: https://github.com/eduNEXT/tutor-contrib-limesurvey/issues/6

### How to test
1. Install this `tutor-contrib-limesurvey` version
2. Run `tutor config save`
3. Run `tutor dev start -d`
4. Create a new LimeSurvey component, click `Edit` and add our remote installation under LimeSurvey URL:
![image](https://github.com/eduNEXT/tutor-contrib-limesurvey/assets/64440265/0358ed04-43ba-47d0-8ef7-a8850aba2814)
You should be able to perform any operation when visiting the LimeSurvey tab from the instructor dashboard.

BryanttV commented:
I'm currently getting a CRSF error at my local:
![image](https://github.com/eduNEXT/tutor-contrib-limesurvey/assets/64033729/c6928b59-1b8d-4b1b-91f8-e8c5fbad2f09)
Also, I can't log in to our remote installation, when I enter the credentials and press the "login" button it just reloads the page.

BryanttV commented on tutorlimesurvey/plugin.py:
```diff
@@ -20,15 +20,16 @@
         ("LIMESURVEY_VERSION", __version__),
-        ("LIMESURVEY_DOCKER_IMAGE", "docker.io/acspri/limesurvey:6.1.0"),
+        ("LIMESURVEY_DOCKER_IMAGE", "docker.io/edunext/limesurvey:15.0.0"),
+        ("LIMESURVEY_DOCKER_BASE_IMAGE", "docker.io/acspri/limesurvey:6.1.0"),
```
Should we start using [this image](https://hub.docker.com/r/adamzammit/limesurvey)?

BryanttV replied:
You made reference to [this repository](https://github.com/adamzammit/limesurvey-docker) in the description of the PR.

Ian2012 replied:
This variable is not used anywhere, does it have any pupose?

felipemontoya replied:
I suppose that is what we use to build our own, right?

Also, the image itself makes no difference. It is published by the same person in both places.

but we should move to 6.2.0 which is now available. https://hub.docker.com/layers/acspri/limesurvey/6.2.0/images/sha256-7020d0a3e849726efe400ace3d27e2fc270d77bb7cb914b8f2e8b97b8edf44b6?context=explore

mariajgrimaldi replied:
Yes, we built the production image on top of it. Thanks!

mariajgrimaldi commented:
@BryanttV addressing your comments:

> I'm currently getting a CRSF error at my local: ![image](https://user-images.githubusercontent.com/64033729/258141626-c6928b59-1b8d-4b1b-91f8-e8c5fbad2f09.png) 

I'm not sure what's going on here, I'll check that out :)

> Also, I can't log in to our remote installation, when I enter the credentials and press the "login" button it just reloads the page.
It's working for me even in a private window, not sure if it's something related to your browser. Can you try removing cookies and stuff?

mariajgrimaldi commented:
These are the test cases I'm currently considering @BryanttV:
1. Login to LimeSurvey service outside the iframe
2. Login to LimeSurvey service inside from the instructor dashboard iframe
In local and remote environments, both of them using the `docker.io/edunext/limesurvey:15.0.0` with the current `config.php` configurations.

Can you specify where it's not working for you? With all the needed details: environment and test case, be as specific as you can so I can address those issues. Thanks!

mariajgrimaldi commented:
I have a theory about this working in dev environments:

BryanttV commented:
According to my tests I obtained the following:

#### local environment
1. **Outside the iframe**: I'm getting a CSRF token error when I log in.
2. **Inside the instructor dashboard iframe:** I'm getting the same CSRF token error when I log in.

#### remote environment
1. **Outside the iframe:** I can log in without any problem.
2. **Inside the instructor dashboard iframe:** I can't log in. When I enter the credentials and press the "login" button it just reloads the page.

mariajgrimaldi commented:
@BryanttV can you test the last changes I uploaded for the local environment? We're now using the same image as before for dev environments. Check out why [here](https://github.com/eduNEXT/tutor-contrib-limesurvey/pull/9#issuecomment-1664187587).

Now, regarding the error in the instructor dashboard iframe, did you try testing it in a private window?

BryanttV commented:
Perfect!
1. Now it works in the local environment, both in the iframe of the instructor's panel and outside of it.
2. It's also working in the remote environment, I just had to use a different browser (firefox)

Ian2012 commented on tutorlimesurvey/plugin.py:
```diff
@@ -20,15 +20,17 @@
         # Prefix your setting names with 'LIMESURVEY_'.
         ("LIMESURVEY_VERSION", __version__),
-        ("LIMESURVEY_DOCKER_IMAGE", "docker.io/acspri/limesurvey:6.1.0"),
+        ("LIMESURVEY_DOCKER_IMAGE", "docker.io/edunext/limesurvey:15.0.0-tmp.0.1"),
```
Can you use `DOCKER_IMAGE_LIMESURVEY` ?

mariajgrimaldi replied:
I thought about it but the in-line comments (from the cookie cutter) say `# Prefix your setting names with 'LIMESURVEY_'`

Ian2012 commented on tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php:
Can you leave a source for this file so in the future maintainers will now how to verify this file?

mariajgrimaldi replied:
That's a good idea! Sure do

BryanttV commented:
LGTM

felipemontoya commented on tutorlimesurvey/templates/limesurvey/build/limesurvey/config.php:
```diff
@@ -0,0 +1,102 @@
+        ),
+        {%- raw %}
+        'session' => array (
+			'sessionName'=>'LS-VBXZZORFJZCAKGYI',
```
Is it reasonable to leave a fixed session name?

Is this something that should be auto generated?

mariajgrimaldi replied:
I copied this from the docs: https://manual.limesurvey.org/Optional_settings#Allow_usage_of_session_and_Csrf_Validation_in_iFrame. It's something I prefer to leave as is.
